crates/oxc_linter/src/rules/eslint/no_multi_str.rs

```diff
@@ -0,0 +1,74 @@
+use lazy_static::lazy_static;
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use regex::Regex;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_multi_str_diagnostic(span0: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("eslint(no-multi-str): .").with_label(span0)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoMultiStr;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow multiline strings.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Some consider this to be a bad practice as it was an undocumented feature of JavaScript
+    /// that was only formalized later.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// var x = "Line 1 \
+    ///  Line 2";
+    /// ```
+    NoMultiStr,
+    style,
+);
+
+impl Rule for NoMultiStr {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        // https://github.com/eslint/eslint/blob/9e6d6405c3ee774c2e716a3453ede9696ced1be7/lib/shared/ast-utils.js#L12
+        lazy_static! {
+            static ref MULTILINE_REGEX: Regex = Regex::new(r"\r\n|[\r\n\u2028\u2029]").unwrap();
+        }
+
+        if let AstKind::StringLiteral(literal) = node.kind() {
+            let source = literal.span.source_text(ctx.source_text());
+            if MULTILINE_REGEX.is_match(source) {
```

Comment:
Shall we remove the Regex? We prefer avoid the regex crate when possible.

https://oxc.rs/docs/contribute/rules.html#development-policy

Reply:
Is it really slower then doing `.contains` and later `.find()` again to find the `\`?

Reply:
I test on my machine with code, It's only a small diff:

```rust
if source.chars().any(|ch| {
    matches!(ch, '\r' | '\n' | '\u{2028}' | '\u{2029}' )
}) {
    ctx.diagnostic(no_multi_str_diagnostic(literal.span));
}
```

For vscode repo

```
hyperfine -w 5 -i -n oxlint_with_reg "./oxlint -A all -D no-multi-str" -n oxlint_with_contains "./oxlint_contains -A all -D no-multi-str"

oxlint_with_reg ran
    1.01 ± 0.02 times faster than oxlint_with_contains

```

For only `bignewlines.js`

```
hyperfine -w 5 -i -n oxlint_with_reg "./oxlint -A all -D no-multi-str bignewlines.js" -n oxlint_with_contains "./oxlint_contains -A all -D no-multi-str bignewlines.js"

 oxlint_with_reg ran
    1.02 ± 0.03 times faster than oxlint_with_contains

```

With real testing, I also this it's OK to use RegExp in this rule :)